src/test/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImplTest.java

```diff
@@ -17,44 +22,125 @@
+  }
+
+  @Test
+  void updateFormDataTestWhileFormIdIsEmpty() {
+    LinkedHashMap<String, Object> map = new LinkedHashMap<>();
+    map.put(STRING, STRING);
+    FormDataSchema formDataSchema = new FormDataSchema(TEST_ID, TEST_FORM_ID, TEST_VERSION, TEST_FORM_DATA, TEST_FORM_META_DATA);
+    Assertions.assertThrows(InvalidInputException.class, () -> formDataService.updateFormData(formDataSchema, "formData.name:akhil", null, null));
+  }
+
+  @Test
+  void updateFormDataTestToSaveToAuditException() throws JsonProcessingException {
+    Map<String, Object> testFormMetaData = new HashMap<>();
+    testFormMetaData.put(FORM_VERSION, 1);
+    Map<String, Object> testFormData = new HashMap<>();
+    testFormData.put(NAME, NAME_VALUE);
+    testFormData.put(AGE,AGE_VALUE);
+    LinkedHashMap<String, Object> map = new LinkedHashMap<>();
+    map.put(STRING, STRING);
+    FormDataSchema formDataSchema = new FormDataSchema(TEST_ID, TEST_FORM_ID, TEST_VERSION, testFormData, testFormMetaData);
+    FormDataDefinition formDataDefinition= new FormDataDefinition();
+    formDataDefinition.setId(TEST_ID);
+    formDataDefinition.setFormId(TEST_FORM_ID);
+    formDataDefinition.setVersion(TEST_VERSION);
+    formDataDefinition.setFormData(testFormData);
+    formDataDefinition.setFormMetaData(testFormMetaData);
+    when(mongoTemplate.findOne(any(),any(),any())).thenReturn(formDataDefinition);
+    when(formDataAuditService.saveFormDataAudit(any())).thenThrow(JsonProcessingException.class);
+    Assertions.assertThrows(InvalidInputException.class,()->formDataService.updateFormData(formDataSchema, "formData.name:akhil", null, null));
+  }
+
+  @Test
+  void updateFormDataTestToSaveToAudit() throws JsonProcessingException {
+    Map<String, Object> testFormMetaData = new HashMap<>();
+    testFormMetaData.put(FORM_VERSION, 1);
+    Map<String, Object> testFormData = new HashMap<>();
+    testFormData.put(NAME, NAME_VALUE);
+    testFormData.put(AGE,AGE_VALUE);
+    LinkedHashMap<String, Object> map = new LinkedHashMap<>();
+    map.put(STRING, STRING);
+    FormDataAuditSchema formDataAuditSchema = new FormDataAuditSchema(TEST_ID,NULL,EMPTY_STRING, TEST_VERSION, testFormData, testFormMetaData);
+    FormDataSchema formDataSchema = new FormDataSchema(TEST_ID, TEST_FORM_ID, TEST_VERSION, testFormData, testFormMetaData);
+    FormDataDefinition formDataDefinition= new FormDataDefinition();
+    formDataDefinition.setId(TEST_ID);
+    formDataDefinition.setFormId(TEST_FORM_ID);
+    formDataDefinition.setVersion(TEST_VERSION);
+    formDataDefinition.setFormData(testFormData);
+    formDataDefinition.setFormMetaData(testFormMetaData);
+    when(mongoTemplate.findOne(any(),any(),any())).thenReturn(formDataDefinition);
+    FormDataAuditResponse formDataAuditResponse=new FormDataAuditResponse(formDataDefinition.getId(),formDataDefinition.getVersion()+1);
+    when(formDataAuditService.saveFormDataAudit(any())).thenReturn(formDataAuditResponse);
```

Comment:
Instead of mocking the response, capture the input to this call and verify the value.